src/main/java/org/json/JSONArray.java

```diff
@@ -177,6 +177,35 @@ public JSONArray(Collection<?> collection) {
         }
     }
 
+    /**
+     * Construct a JSONArray from an Iterable. This is a shallow copy.
+     *
+     * @param iter
+     *            A Iterable collection.
+     */
+    public JSONArray(Iterable<?> iter) {
+        this();
+        if (iter == null) {
+            return;
+        }
+        this.addAll(iter);
+    }
+
+    /**
+     * Construct a JSONArray from another JSONArray. This is a shallow copy.
+     *
+     * @param collection
+     *            A Collection.
+     */
+    public JSONArray(JSONArray array) {
```

Comment:
Do we support this for JSONObject? If not, we should not support it for JSONArray either.

Reply:
After considering, withdrawing the constructor comments. Should be fine to add these methods.